src/nomad_simulations/properties/band_structure.py

```diff
@@ -0,0 +1,377 @@
+        repeats=True,
+        description="""
+        Contributions to the electronic eigenvalues. Example, in the case of a DFT+GW calculation, the GW eigenvalues
+        are stored under `value`, and each contribution is identified by `label`:
+            - `'KS'`: Kohn-Sham contribution. This is also stored in the DFT entry under `ElectronicEigenvalues.value`.
+            - `'KSxc'`: Diagonal matrix elements of the expectation value of the Kohn-Sahm exchange-correlation potential.
+            - `'SigX'`: Diagonal matrix elements of the exchange self-energy. This is also stored in the GW entry under `ElectronicSelfEnergy.value`.
+            - `'SigC'`: Diagonal matrix elements of the correlation self-energy. This is also stored in the GW entry under `ElectronicSelfEnergy.value`.
+            - `'Zk'`: Quasiparticle renormalization factors contribution. This is also stored in the GW entry under `QuasiparticleWeights.value`.
+        """,
+    )
+
+    reciprocal_cell = Quantity(
+        type=KSpace.reciprocal_lattice_vectors,
+        description="""
+        Reference to the reciprocal lattice vectors stored under `KSpace`. This reference is useful when resolving the Brillouin zone
+        for the front-end visualization.
+        """,
+    )
+
+    def __init__(
+        self, m_def: Section = None, m_context: Context = None, **kwargs
+    ) -> None:
+        super().__init__(m_def, m_context, **kwargs)
+        self.name = self.m_def.name
+
+    def validate_occupation(self, logger: BoundLogger) -> bool:
+        """
+        Validate `occupation` by checking if they exist and have the same shape as `value`.
+
+        Args:
+            logger (BoundLogger): The logger to log messages.
+
+        Returns:
+            (bool): True if the shape of `occupation` is the same as `value`, False otherwise.
+        """
+        if self.occupation is None or len(self.occupation) == 0:
+            logger.warning('Cannot find `occupation` defined.')
+            return False
+        if self.value is not None and self.value.shape != self.occupation.shape:
+            logger.warning(
+                'The shape of `value` and `occupation` are different. They should have the same shape.'
+            )
+            return False
+        return True
+
+    def order_eigenvalues(
+        self, logger: BoundLogger
+    ) -> Tuple[Optional[pint.Quantity], Optional[np.ndarray]]:
+        """
+        Order the eigenvalues based on the `value` and `occupation`. The return `value` and
+        `occupation` are flattened.
+
+        Args:
+            logger (BoundLogger): The logger to log messages.
+
+        Returns:
+            (Tuple[Optional[pint.Quantity], Optional[list]]): The flattened and sorted `value` and `occupation`.
+        """
+
+        # Check if `value` exists
+        if self.value is None or len(self.value) == 0:
+            logger.error('Could not find `value` defined.')
+            return None, None
```

Comment:
Like here: this teste isn't necessary if you maintain arrays as the output.
Moreover, shouldn't instantiation assure that only 1 condition has to be met?

Reply:
Well, someone could call for this method without `value` set up. Now, for the `is None` and `len()`, it is true that the length check does not make sense, so I will remove it.